### DISCUSSION TITLE
Custom Eye on the Animated Eye example.

### DISCUSSION THREAD

**Discussion by BishoPower** [Bodmer/TFT_eSPI#1454] [Q&A - General]
Can anyone give me some pointers on going about creating a custom eye for the animated eye example?

**Comment by Bodmer** [comment]
You need to create a set of image files or adapt the ones provided. Example images are here:
https://github.com/adafruit/Uncanny_Eyes/tree/master/convert/defaultEye

The python script to convert these images to a header file for inclusion in the sketch are here.
https://github.com/adafruit/Uncanny_Eyes/tree/master/convert

The sketch is adapted from one by Adafruit, so you may find help on the Adafruit website:
https://learn.adafruit.com/animated-electronic-eyes/customizing

**Reply by jabural** [reply]
About generating new images, I don't know why, but when I generate a header file using the tablegen.py and the add it to the data in the animated eyes file, it doesn't appear anything on the screens when I upload it. I don't change the images and the command that I use to generate the image file is the same as the one in the adafruit website:

python tablegen.py defaultEye/sclera.png defaultEye/iris.png defaultEye/lid-upper-symmetrical.png defaultEye/lid-lower-symmetrical.png defaultEye/lid-upper.png defaultEye/lid-lower.png 80 > defaultEye.h

So I don't know if there is something different that I should do for generating eyes for the esp32, because for me it appears that the file generated is the same. But also if I try to use the graphics header file in the teensy project, it doesn't work, so I imagine is that there is something different for the files for the esp32.

Thanks in advance for your answer

**Reply by keithB105** [reply]
I've tried and came up with the same results. The generator program isn't setup to support the ESP32 and 240x240 round displays. It appears that it's coded for the 128x128 display and if I'm reading it right, it's also setting some bits at 8bit and that is too low for the 240x240 displays. Another note, when I compiled the code as listed, it doesn't seem to create all the hex file arrays for each of the images and the ones that it does, it's not the same size as what they provide. It's very odd and I'm really outside my comfort zone on this. If anyone out there has a working copy for the eyes for the ESP32 and 240x240 round display. I would be thrilled to hear about it.... otherwise I will keep trying to figure things out.

**Comment by Bodmer** [comment]
Have a look here also as someone has created a custom eye and describes the process:
https://makezine.com/projects/eye-of-newt-keep-watch-with-a-creepy-compact-animated-eyeball/

**Comment by labomat** [comment]
Had the same problem. I used the image files from https://github.com/PaulStoffregen/ST7735_t3/blob/master/examples/uncannyEyes_async_st7789_240x240/graphics/default_large.h but got a blank screen as well.
